### PR TITLE
Hotfix for 3.3.03 Update Faction Data

### PR DESCRIPTION
**Pull request by Hoodathunk**
Elite Dangerous 3.3.03 update now provides system & station controlling faction data (name & state) as an object.

### CODE REVIEW

File: JournalMonitor/JournalMonitor.cs [hunk 1]
```diff
@@ -139,7 +138,7 @@ public static List<Event> ParseJournalEntry(string line, bool fromLogLoad = fals
                                     bool wanted = JsonParsing.getOptionalBool(data, "Wanted") ?? false;
                                     bool activeFine = JsonParsing.getOptionalBool(data, "ActiveFine") ?? false;
 
-                                    events.Add(new DockedEvent(timestamp, systemName, systemAddress, marketId, stationName, stationState, stationModel, allegiance, faction, factionState, Economies, government, distancefromstar, stationServices, cockpitBreach, wanted, activeFine) { raw = line, fromLoad = fromLogLoad });
+                                    events.Add(new DockedEvent(timestamp, systemName, systemAddress, marketId, stationName, stationState, stationModel, allegiance, faction.name, faction.FactionState, Economies, government, distancefromstar, stationServices, cockpitBreach, wanted, activeFine) { raw = line, fromLoad = fromLogLoad });
```
Comment (Tkael) [inline JournalMonitor/JournalMonitor.cs]: Consider passing faction as an object? We're reconstituting the object at EDDI.cs line 1027. We can simplify if we build the object here and simply pass the object.

File: JournalMonitor/JournalMonitor.cs [hunk 2]
```diff
@@ -217,7 +215,7 @@ public static List<Event> ParseJournalEntry(string line, bool fromLogLoad = fals
                                         factions = getFactions(factionsVal);
                                     }
 
-                                    events.Add(new JumpedEvent(timestamp, systemName, systemAddress, x, y, z, starName, distance, fuelUsed, fuelRemaining, boostUsed, allegiance, faction, factionState, economy, economy2, government, security, population, factions) { raw = line, fromLoad = fromLogLoad });
+                                    events.Add(new JumpedEvent(timestamp, systemName, systemAddress, x, y, z, starName, distance, fuelUsed, fuelRemaining, boostUsed, allegiance, faction.name, faction.FactionState, economy, economy2, government, security, population, factions) { raw = line, fromLoad = fromLogLoad });
```
Comment (Tkael) [inline JournalMonitor/JournalMonitor.cs]: Consider passing faction as an object? We're reconstituting the object at EDDI.cs line 1280. We can simplify if we build the object here and simply pass the object.

File: JournalMonitor/JournalMonitor.cs [hunk 3]
```diff
@@ -266,7 +264,7 @@ public static List<Event> ParseJournalEntry(string line, bool fromLogLoad = fals
                                         factions = getFactions(factionsVal);
                                     }
 
-                                    events.Add(new LocationEvent(timestamp, systemName, x, y, z, systemAddress, body, bodyType, docked, station, stationtype, marketId, allegiance, faction, economy, economy2, government, security, population, longitude, latitude, factions) { raw = line, fromLoad = fromLogLoad });
+                                    events.Add(new LocationEvent(timestamp, systemName, x, y, z, systemAddress, body, bodyType, docked, station, stationtype, marketId, allegiance, faction.name, faction.FactionState, economy, economy2, government, security, population, longitude, latitude, factions) { raw = line, fromLoad = fromLogLoad });
```
Comment (Tkael) [inline JournalMonitor/JournalMonitor.cs]: Consider passing faction as an object? We're reconstituting the object at EDDI.cs line 926. We can simplify if we build the object here and simply pass the object.

File: Events/LocationEvent.cs [hunk 4]
```diff
@@ -59,6 +59,8 @@ static LocationEvent()
 
         public string faction { get; private set; }
 
+        public string factionstate => (factionState ?? FactionState.None).localizedName;
```
Comment (richardbuckle) [inline Events/LocationEvent.cs]: Not too keen on the subtle case distinction between `factionstate` and `factionState`. I suppose it's for Cottle?

**Pull request merged**